packages/ts/react-grid/test/autogrid.spec.tsx

```diff
@@ -126,26 +100,138 @@ describe('@hilla/react-grid', () => {
+        firstNameFilterField.dispatchEvent(new CustomEvent('input'));
+
+        const expectedPropertyFilter: PropertyStringFilter = {
+          ...{ t: 'propertyString' },
+          filterValue: 'filter-value',
+          propertyId: 'firstName',
+          matcher: Matcher.CONTAINS,
+        };
+        const expectedFilter: AndFilter = { ...{ t: 'and' }, children: [expectedPropertyFilter] };
+        expect(personService.lastFilter).to.eql(expectedFilter);
       });
+      it('combine filters (and) when you type in multiple fields', async () => {
+        const result = render(<TestAutoGrid headerFilters />);
+        await nextFrame();
+        await nextFrame();
+        const grid: GridElement = result.container.querySelector('vaadin-grid')!;
+        const firstNameFilterField = getHeaderCellContent(grid, 1, 0).firstElementChild as TextFieldElement;
+        firstNameFilterField.value = 'filterFirst';
+        firstNameFilterField.dispatchEvent(new CustomEvent('input'));
+
+        const lastNameFilterField = getHeaderCellContent(grid, 1, 1).firstElementChild as TextFieldElement;
+        lastNameFilterField.value = 'filterLast';
+        lastNameFilterField.dispatchEvent(new CustomEvent('input'));
 
-      it('should ignore unknown columns', () => {
-        const result = render(<TestAutoGrid visibleColumns={['foo', 'email', 'bar', 'firstName']} />);
-        const columns = result.container.querySelectorAll('vaadin-grid-sort-column');
-        expect(columns.length).to.equal(2);
-        expect(columns[0].path).to.equal('email');
-        expect(columns[0].header).to.equal('Email');
-        expect(columns[1].path).to.equal('firstName');
-        expect(columns[1].header).to.equal('First name');
+        const expectedFirstNameFilter: PropertyStringFilter = {
+          ...{ t: 'propertyString' },
+          filterValue: 'filterFirst',
+          propertyId: 'firstName',
+          matcher: Matcher.CONTAINS,
+        };
+        const expectedLastNameFilter: PropertyStringFilter = {
+          ...{ t: 'propertyString' },
+          filterValue: 'filterLast',
+          propertyId: 'lastName',
+          matcher: Matcher.CONTAINS,
+        };
+        const expectedFilter: AndFilter = {
+          ...{ t: 'and' },
+          children: [expectedFirstNameFilter, expectedLastNameFilter],
+        };
+        expect(personService.lastFilter).to.eql(expectedFilter);
+      });
+      it('removes filters if turning header filters off', async () => {
+        const result = render(<AutoGrid service={personService} model={PersonModel} headerFilters></AutoGrid>);
+        await nextFrame();
+        await nextFrame();
+        result.rerender(<AutoGrid service={personService} model={PersonModel}></AutoGrid>);
+        await nextFrame();
+        await nextFrame();
+        const grid: GridElement = result.container.querySelector('vaadin-grid')!;
+        expect(getHeaderCellContent(grid, 0, 0).innerText).to.equal('First name');
```

Comment:
Should it also clear the filter state and reload from data provider?

There's also some weirdness if you enter a filter value and then toggle header filters. The grid will still be filtered, but the filter fields are empty because we don't initialize them from the initial filter. Maybe resetting filter state when disabling header filters is easiest for now.

Reply:
Sounds reasonable to remove all filters if you remove the filter row